tests/amor/pipeline_test.py

```diff
@@ -0,0 +1,61 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2024 Scipp contributors (https://github.com/scipp)
+# flake8: noqa: F403, F405
+
+import numpy as np
+import pytest
+import sciline
+import scipp as sc
+from orsopy import fileio
+
+from essreflectometry import orso
+from essreflectometry.amor import default_parameters
+from essreflectometry.amor import orso as amor_orso
+from essreflectometry.amor import providers
+from essreflectometry.types import *
+
+
+@pytest.fixture()
+def amor_pipeline() -> sciline.Pipeline:
+    params = {
+        **default_parameters,
+        QBins: sc.geomspace(
+            dim='Q', start=0.008, stop=0.075, num=200, unit='1/angstrom'
+        ),
+        SampleRotation[Sample]: sc.scalar(0.7989, unit='deg'),
+        Filename[Sample]: "sample.nxs",
+        SampleRotation[Reference]: sc.scalar(0.8389, unit='deg'),
+        Filename[Reference]: "reference.nxs",
+        WavelengthEdges: sc.array(
+            dims=['wavelength'], values=[2.4, 16.0], unit='angstrom'
+        ),
+        orso.OrsoCreator: orso.OrsoCreator(
+            fileio.base.Person(
+                name='Max Mustermann',
+                affiliation='European Spallation Source ERIC',
+                contact='max.mustermann@ess.eu',
+            )
+        ),
+    }
+    return sciline.Pipeline(
+        (*providers, *orso.providers, *amor_orso.providers), params=params
+    )
+
+
+def test_run_pipeline(amor_pipeline: sciline.Pipeline):
+    res = amor_pipeline.compute(orso.OrsoIofQDataset)
+    assert res.info.data_source.experiment.instrument == 'AMOR'
+    assert res.info.reduction.software.name == 'ess.reflectometry'
+    assert res.data.ndim == 2
+    assert res.data.shape[1] == 4
+    assert np.all(res.data[:, 1] > 0)
+
+
+def test_find_corrections(amor_pipeline: sciline.Pipeline):
+    graph = amor_pipeline.get(orso.OrsoIofQDataset)
+    assert sorted(orso.find_corrections(graph)) == [
+        'chopper ToF correction',
+        'footprint correction',
+        'supermirror calibration',
+        'total counts',
+    ]
```

Comment:
I wonder if the order is relevant, or if ORSO tracks it?

Reply:
So you would have to somehow start from the end point, walk the graph backwards to build a list of corrections, and then save that list in reverse order in the orso file, to make sure the order in which the corrections were applied is preserved?

Reply:
Yeah, that makes the most sense. For ORSO, the corrections is just a list, so easy enough to reverse.

Reply:
The order can in some cases be undefined if two or more correction steps are independent and happen 'at the same time'. But using topological order of the task graph would do the trick. (See naive scheduler). Not the cheapest solution but cheap enough here.